hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+
+        all_fields = [
+            'company_type', 'name', 'phone', 'street_number', 'street_number2',
+            'email', 'street', 'street2', 'city', 'zip', 'state_id', 'country_id',
+            'vat', 'l10n_mx_edi_curp', 'website', 'lang', 'category_id', 'l10n_mx_edi_fiscal_regime', 'parent_id'
+        ]
+
+        # Construir el diccionario contact_data con solo los campos enviados en el JSON
+        contact_data = {field: request.jsonrequest.get(field) for field in all_fields if request.jsonrequest.get(field) is not None}
+
+        # Si category_id está presente, convertirlo en una lista
+        if 'category_id' in contact_data and not isinstance(contact_data['category_id'], list):
+            contact_data['category_id'] = [contact_data['category_id']]
+        
+        list_not_title = ['lang', 'company_type', 'email', 'website']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()
+
+        contact_id = request.env['res.partner'].create(contact_data)
+
+        # Utilizar logger en lugar de print
+        logger.info("Nuevo contacto creado con ID %s", contact_id.id)
+        
+        return {"message": f"Nuevo contacto creado con ID: {contact_id.id}", "contact_id": contact_id.id}
+    
+    @route('/update_contact', methods=['POST'], type='json', auth='user')
+    def update_contact(self):
+
+        data = request.jsonrequest
+        partner_id = data.get('partner_id')  # ID del contacto a actualizar
+        update_vals = data.get('update_vals')  # Valores a actualizar
+
+        # Encontrar el contacto por su ID
+        partner = request.env['res.partner'].browse(partner_id)
+
+        list_not_title = ['lang', 'company_type', 'email', 'website', 'vat']
+
+        for key, value in update_vals.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    update_vals[key] = value.title()
+
+        # Actualizar el contacto con los nuevos valores
+        partner.write(update_vals)
+
+        logger.info("Contacto actualizado con ID %s", partner_id)
+
+        return {"message": f"Contacto con ID: {partner_id} actualizado exitosamente."}
+    
+    @route('/manage_contact', methods=['POST'], type='json', auth='user')
+    def manage_contact(self):
+        
+        data = request.jsonrequest
+        email = data.get('email')
+        phone = data.get('phone')
+        contact_data = data.get('contact_data', {})  # Datos del contacto a actualizar/crear
+
+        # Verificar si ya existe un contacto a partir de los primeros digitos del correo y ultimos digitos del telefono
+        email_prefix = email[0:5]
+        phone_suffix = phone[len(phone)-5:len(phone)]
+
+        # Crear dominios de búsqueda usando patrones con comodines
+        domain = [('email', 'ilike', f'{email_prefix}%'), ('phone', 'ilike', f'%{phone_suffix}')]
```

Comment:
why no you use the ids?

I mean, you site are creating contacts and returning it's ids to allow post identification

Reply:
Because there are already contacts in odoo that we want to know if they exist before creating them